### ISSUE TITLE
Split docker-compose into local and deployed version

### ISSUE THREAD

**Issue by katherine-sussol** [openmsupply/unified-codes#151]
## Current Behaviour

Only one config which sets up everything

## Possible Solution

- Local version to spin up dgraph and keycloak only 
  - 2 containers now, eventually this will be 4 containers (3 for dgraph and 1 for keycloak)
- Deployed file to spin up everything 
  - dgraph
  - keycloak
  - legacy-api
  - data-service

**Comment by wlthomson** [comment]
Does deployed version here mean production build?

Possibly related to #149 (setup configs etc. as part of production build).

**Comment by katherine-sussol** [comment]
Hmm so that's a good question, I'm not really sure yet what the plan is r.e. environments - is it just going to be local dev and production? I guess what I really mean by deployed is anything that's not our local environment, (e.g. ci/staging/prod)

**Comment by wlthomson** [comment]
I also haven't thought too much about this (I'll draw up a diagram of my current thinking, maybe that will help)!

**Comment by wlthomson** [comment]
Just a quick diagram of my interpretation of the architecture we're currently thinking of:

![unified-codes-architecture-v1](https://user-images.githubusercontent.com/43223637/91542001-4798e480-e971-11ea-8820-b0e6223713ac.jpg)

Because the docker local network ports will be different from what we would use in debugging (or in a production environment without a docker network), I was thinking of debug v. production v. docker. In general, we won't be trying out any non-dockerised production environments (at least not now?), so I think we only need to think about debug and docker (which can be conceptualised as local and deployed, maybe?).

Don't know if that helps at all?

**Comment by katherine-sussol** [comment]
Yeah that was kind of my thinking too, r.e. Debug and Docker, at least for now. 
If there are any other dockerised environments needed in future (e.g. for staging maybe) it would just be a matter of creating another .env file for it and pointing to it inside the 'production' version of docker-compose file (i.e. not the debug one), which should be straightforward enough?

Nice diagram, matches what I was seeing! (I think possible extension would be persistent storage for auth service, and external access to data-service? Or better to close that off? Definitely agree with Dgraph endpoints closed to public)

**Comment by wlthomson** [comment]
> Nice diagram, matches closely what I was seeing in terms of deployment - (I think possible extension would be persistent storage for auth service, and external access to data-service? Or better to close that off? Definitely agree with Dgraph endpoints closed to public)

Yeah, `auth-service` should also be connected to persistent storage (couldn't neatly fit it into the diagram, so just omitted!). External access to the `data-service` is also a good point, e.g. `http://umc.org/graphql` perhaps?
